contrib/natsio/src/test/java/com/google/cloud/dataflow/contrib/natsio/NatsIOTest.java

```diff
@@ -0,0 +1,145 @@
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.KV;
+
+public class NatsIOTest implements Serializable {
+	private static final Logger LOG = LoggerFactory.getLogger(NatsIOBench.class);	
+	
+	private String servers;
+	private String queue;
+	
+	private String project;
+	private String stagingLocation;
+	private int maxRecords;
+	private int maxReadtime;
+	private int loop;
+	private int interval;
+	private int consumers;
+	private int producers;
+	private List<String> subjects;
+	private String subject;
+	
+	private Properties props;
+	
+	public NatsIOTest() {
+		servers = System.getProperty("nats.servers");
+		queue = System.getProperty("nats.queue");
+
+		project = System.getProperty("project");
+		stagingLocation = System.getProperty("stagingLocation");
+		
+		maxRecords = (System.getProperty("nats.maxRecords") == null ? 30000 : Integer.parseInt(System.getProperty("nats.maxRecords")));
+		maxReadtime = (System.getProperty("nats.maxReadtime") == null ? 100 : Integer.parseInt(System.getProperty("nats.maxReadtime")));
+
+		subjects = (System.getProperty("subjects") == null ? Arrays.asList("test") : Arrays.asList(System.getProperty("subjects").split(",")));
+		subject =  subjects.get(0);
+		
+		loop = (System.getProperty("loop") == null ? 30000 : Integer.parseInt(System.getProperty("loop")));
+		interval = (System.getProperty("interval") == null ? 0 : Integer.parseInt(System.getProperty("interval")));
+		consumers = (System.getProperty("consumers") == null ? 1 :Integer.parseInt(System.getProperty("consumers")));
+		producers = (System.getProperty("producers") == null ? 1 : Integer.parseInt(System.getProperty("producers")));
+		
+		props = new Properties();
+		props.setProperty("servers", servers);
+		if (queue != null)
+			props.setProperty("queue", queue);
+	}
+	
+	
+	private BlockingDataflowPipelineOptions buildOptions() {
+		BlockingDataflowPipelineOptions options = PipelineOptionsFactory.create().as(BlockingDataflowPipelineOptions.class);
+		options.setRunner(BlockingDataflowPipelineRunner.class);
+		options.setProject(project);
+		options.setStagingLocation(stagingLocation);
+		options.setNumWorkers(1);
+		
+		return options;
+	}
+	
+	@Test
+	public void simplePublish() {
+		BlockingDataflowPipelineOptions options = buildOptions();
+		options.setJobName("simple-publish-" + System.currentTimeMillis());
+		
+		Pipeline p = Pipeline.create(options);
+		p
+			.apply(Create.of(subject)).setCoder(StringUtf8Coder.of())
+			.apply(ParDo.of(new DoFn<String, KV<String,String>>() {
+				@Override
+				public void processElement(DoFn<String, KV<String, String>>.ProcessContext c) throws Exception {
+					c.output(KV.of(c.element(), "hello world!!!"));
+				}
+			}))
+			.apply(NatsIO.Write.to(props));
+		p.run();
+	}
+	
+	@Test
+	public void publishSubscribe() throws InterruptedException {
```

Comment:
I don't understand why these are `@Test` unit tests instead of standalone executables we run. Compare to `DataflowWordCount` --  this is in an `examples` folder.

Unit tests, by definition, should have no external dependencies.

Similarly, I should be able to run `mvn install` with no arguments and no network, and this should succeed (as long as I've downloaded all the .jars the project depends on).